mne/stats/cluster_level.py

```diff
@@ -557,11 +557,20 @@ def _cluster_indices_to_mask(components, n_tot):
     return components
 
 
-def _cluster_mask_to_indices(components):
+def _cluster_mask_to_indices(components, shape):
     """Convert to the old format of clusters, which were bool arrays."""
     for ci, c in enumerate(components):
-        if not isinstance(c, slice):
-            components[ci] = np.where(c)[0]
+        if isinstance(c, np.ndarray):  # mask
+            components[ci] = np.where(c.reshape(shape))
+        else:
+            assert isinstance(c, tuple), type(c)
+            c = list(c)  # tuple->list
+            for ii, cc in enumerate(c):
+                if isinstance(cc, slice):
+                    c[ii] = np.arange(cc.start, cc.stop)
```

Comment:
doesn't np.arange return a float? do you caste it to int somewhere?

Reply:
only if you use float arguments
```
>>> np.arange(1, 3)
array([1, 2])
>>> np.arange(1, 3).dtype
dtype('int64')
```
start and step here are int so it should be fine